api/routes/user.py

```diff
@@ -18,3 +22,42 @@ def get_current_user_profile(
 
     log(log.INFO, f"User {current_user.username} requested his profile")
     return current_user
+
+
+@user_router.patch("/me", status_code=status.HTTP_200_OK, response_model=s.User)
+def update_user_username(
+    user_update: s.UserUpdate,
+    current_user: m.User = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Updates the current user's info"""
+    if user_update.alias is not None and current_user.alias != user_update.alias:

```

Comment:
`current_user(**user_update.model_validate(exclude_none=True)`
Please try this instead of if statements